packages/zowe-explorer/src/trees/dataset/DatasetTree.ts

```diff
@@ -113,136 +112,278 @@ export class DatasetTree extends ZoweTreeProvider<IZoweDatasetTreeNode> implemen
-                    encodingInfo?.kind === "binary" ? "&encoding=binary" : encodingInfo?.kind === "other" ? "&encoding=" + encodingInfo?.codepage : ""
-                }`;
-                await DatasetFSProvider.instance.writeFile(
-                    destUri.with({
-                        query: queryString,
-                    }),
-                    contents,
-                    { create: true, overwrite: true }
-                );
+                const entry = await DatasetFSProvider.instance.fetchDatasetAtUri(destUri);
+                if (entry == null) {
+                    await destApi.createDataSet(
+                        zosfiles.CreateDataSetTypeEnum.DATA_SET_SEQUENTIAL,
+                        dsname,
+                        {}
+                    );
+                }
             } catch (err) {
-                // If the write fails, we cannot move to the next file
-                if (err instanceof Error) {
+                const code = err?.errorCode?.toString();
+                if (code === "404" || code === "500") {
                     Gui.errorMessage(vscode.l10n.t("Failed to move {0}: {1}", dsname, err.message));
+                    return;
                 }
-                return;
             }
+        }
+
+        // ─────────────────────────────
+        // Transfer + delete
+        // ─────────────────────────────
+        const contents = await DatasetFSProvider.instance.readFile(sourceNode.resourceUri);
+
+        try {
+            const encodingInfo = await sourceNode.getEncoding();
+            const queryString =
+                `forceUpload=true` +
+                (encodingInfo?.kind === "binary"
+                    ? "&encoding=binary"
+                    : encodingInfo?.kind === "other"
+                        ? "&encoding=" + encodingInfo.codepage
+                        : "");
+
+            await DatasetFSProvider.instance.writeFile(
+                destinationWriteUri.with({ query: queryString }),
+                contents,
+                { create: true, overwrite: true }
+            );
 
             if (!recursiveCall) {
-                // Delete any files from the selection on the source LPAR
                 await vscode.workspace.fs.delete(sourceNode.resourceUri, { recursive: false });
-                //dialog message for successful move of pds and members
-                Gui.infoMessage(vscode.l10n.t("Data set(s) moved successfully."));
+                Gui.infoMessage(vscode.l10n.t("Data moved successfully."));
+            }
+        } catch (err) {
+            if (err instanceof Error) {
+                Gui.errorMessage(vscode.l10n.t("Failed to move {0}: {1}", dsname, err.message));
             }
         }
     }
 
     public async handleDrop(
         targetNode: IZoweDatasetTreeNode | undefined,
         dataTransfer: vscode.DataTransfer,
-        _token: vscode.CancellationToken
+        _token?: vscode.CancellationToken
```

Comment:
The VS Code API doc suggests that `token` is a required parameter:

https://code.visualstudio.com/api/references/vscode-api#TreeDragAndDropController&lt;T&gt;

I realize that we don't do anything with it (we probably should in case user cancels the drop, but that's a separate issue) - but we can remove the `?` here:



```suggestion
        _token: vscode.CancellationToken
```